### ISSUE TITLE
Implement better house sorting

### ISSUE THREAD

**Issue by symroe** [addok/addok#208]
When working on #207 I realised that ordering houses in a meaningful way is a little harder than just number sorting (in the UK context at least). 

Here are some examples of the "building name" field from the UK address data:

```
AKEMAN HOUSE 235-237
ALDRIDGE COURT 176A
QUARRIES HOUSE UNIT 2
YARD 28
X Q 7 BUILDING
VICTORIA HALL TWO
```

It's not clear what the sorting order of these should be, but it seems that when we have a letter prefix (like 176A above), '176A' should come before '176B' or 'C'.

I think `sorter` should be moved out of `shell.py` in to `helpers` somewhere, and the sorter should be able to understand cases like the above.

**Comment by yohanboniface** [comment]
> I think sorter should be moved out of shell.py in to helpers somewhere, and the sorter should be able to understand cases like the above.

Sounds good.
But keep in mind the `shell` module is only for debugging purpose. When using the API, the sorting is of course based on pertinence.

**Comment by symroe** [comment]
Ah, I see.

Is there a case where results in the API are sorted by the house number
field? Or is this always left to the ranking?

**Comment by yohanboniface** [comment]
> Or is this always left to the ranking?

Always left to the ranking.

**Comment by symroe** [comment]
Fair enough. I think this is still something that we should implement, but
I might try to do it in its own project first.
